### PR TITLE
Аdd operation status omit

### PR DESCRIPTION
**Pull request by dadlex**
OMIT will be used to complete an item for whatever purpose that shouldn't be registered anywhere in the metrics and affect them

### CODE REVIEW

File: src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java [hunk 1]
```diff
@@ -303,10 +303,14 @@ public final boolean put(final O opResult) {
...
 			}
 		} else if (Status.PENDING.equals(status)) {
-			//in case driver cannot finish operation due to storage API issues or of some other sort, we need
-			//to set the operation status to Pending, so that we don't count it in the metrics and recycle the operation
+			// in case driver cannot finish operation due to storage API issues or of some other sort, we need
+			// to set the operation status to Pending, so that we don't count it in the metrics and recycle the operation
 			generator.recycle(opResult);
 			counterResults.increment();
+		} else if (Status.OMIT.equals(status)) {
+			// operation status is set to Pending in case we want an operation to complete, but not to register
```
Comment (AndreyKoltsov1997) [inline src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java]: I assume there's a typo in the comment: `Pending` -> `Omit`

File: src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java [hunk 2]
```diff
@@ -384,10 +388,14 @@ public final int put(final List<O> opResults, final int from, final int to) {
...
 				}
 			} else if (Status.PENDING.equals(status)) {
-				//in case driver cannot finish operation due to storage API issues or of some other sort, we need
-				//to set the operation status to Pending, so that we don't count it in the metrics and recycle the operation
+				// in case driver cannot finish operation due to storage API issues or of some other sort, we need
+				// to set the operation status to Pending, so that we don't count it in the metrics and recycle the operation
 				generator.recycle(opResult);
 				counterResults.increment();
+			} else if (Status.OMIT.equals(status)) {
+				// operation status is set to Pending in case we want an operation to complete, but not to register
```
Comment (AndreyKoltsov1997) [inline src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java]: Same typo within the comment here.

**Review by AndreyKoltsov1997 — COMMENTED** [review]
Is there a way user would be aware of omit / pending operations and/or their count? If not, do you think it should be implemented?

File: src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java (hunk 1 above)
Comment (dadlex) [reply src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java]: it was a typo, yes, it's fixed now

File: src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java (hunk 2 above)
Comment (dadlex) [reply src/main/java/com/emc/mongoose/base/load/step/local/context/LoadStepContextImpl.java]: yep

**Comment by dadlex** [comment]
> Is there a way user would be aware of omit / pending operations and/or their count? If not, do you think it should be implemented?

no. it's implementation details, user shouldn't be aware of any of them. only succ/fail ops.

**Review by AndreyKoltsov1997 — APPROVED** [review]
LGTM.

**Review by veronikaKochugova — APPROVED** [review]

**Pull request merged**